File: tutorial/04-recursive/problems/02-fibonacci.py
class SoftP0402:
    def fib(self,n):
        if n ==1 or n==2:
            return 1
        else:
            return self.fib(n-1)+self.fib(n-2)
    def fib2(self,n):
        if n==1:
            return 0,1
        if n==2:
            return 0,1
        if n==3:
            return 1,1
        a,b=self.fib2(n-1)
        return b,a+b

File: tutorial/04-recursive/problems/02-fibonacci.java
class SoftP0402{
    public static int fib(int n){
if(n==1 || n==2)return 1;
return SoftP0402.fib(n-1)+SoftP0402.fib(n-2);
    }
    public static int[] fib2(int n){
        if(n==1)return new int[]{0,1};
       if(n==2)return new int[]{0,1};
       if(n==3)return new int[]{1,1};
       int [] numbers=SoftP0402.fib2(n-1);
       return new int[]{numbers[1],numbers[1]+numbers[0]};
    }
    public static void main(String[] args) {
        System.out.println(SoftP0402.fib(4));
        System.out.println(java.util.Arrays.toString(SoftP0402.fib2(10)));
    }
}